### PR TITLE
Bertscore warnings

### PR DESCRIPTION
**Pull request by baskrahmer**
## What does this PR do?

Fixes a number of warnings when using `BertScore` with the default settings:
```
Some weights of RobertaModel were not initialized from the model checkpoint at roberta-base and are newly initialized: ['roberta.pooler.dense.bias', 'roberta.pooler.dense.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
```

and

```
`resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
```

which I think shouldn't be raised when using the metric correctly.

----
📚 Documentation preview 📚: https://torchmetrics--2626.org.readthedocs.build/en/2626/

### CODE REVIEW

File: src/torchmetrics/functional/text/bert.py [hunk 1]
```diff
@@ -33,6 +35,19 @@
 from torchmetrics.utilities.checks import _SKIP_SLOW_DOCTEST, _try_proceed_with_timeout
 from torchmetrics.utilities.imports import _TQDM_AVAILABLE, _TRANSFORMERS_GREATER_EQUAL_4_4
 
+
+@contextmanager
+def ignore_log_warning():
```
Comment (Borda) [inline src/torchmetrics/functional/text/bert.py:40]:
```suggestion
def _ignore_log_warning():
```

File: src/torchmetrics/functional/text/bert.py [hunk 2]
```diff
@@ -41,8 +56,9 @@
 
     def _download_model_for_bert_score() -> None:
         """Download intensive operations."""
-        AutoTokenizer.from_pretrained(_DEFAULT_MODEL, resume_download=True)
-        AutoModel.from_pretrained(_DEFAULT_MODEL, resume_download=True)
+        with ignore_log_warning():
```
Comment (Borda) [inline src/torchmetrics/functional/text/bert.py:59]:
```suggestion
        with _ignore_log_warning():
```

File: src/torchmetrics/functional/text/bert.py [hunk 3]
```diff
@@ -356,8 +372,9 @@ def bert_score(
                 " `transformers` model are used."
                 f"It is, therefore, used the default recommended model - {_DEFAULT_MODEL}."
             )
-        tokenizer = AutoTokenizer.from_pretrained(model_name_or_path or _DEFAULT_MODEL)
-        model = AutoModel.from_pretrained(model_name_or_path or _DEFAULT_MODEL)
+        with ignore_log_warning():
```
Comment (Borda) [inline src/torchmetrics/functional/text/bert.py:375]:
```suggestion
        with _ignore_log_warning():
```

**Review by Borda — APPROVED** [review]

File: src/torchmetrics/functional/text/bert.py [hunk 4]
```diff
@@ -33,6 +35,19 @@
 from torchmetrics.utilities.checks import _SKIP_SLOW_DOCTEST, _try_proceed_with_timeout
 from torchmetrics.utilities.imports import _TQDM_AVAILABLE, _TRANSFORMERS_GREATER_EQUAL_4_4
 
+
+@contextmanager
+def _ignore_log_warning():  # noqa: ANN202
```
Comment (Borda) [inline src/torchmetrics/functional/text/bert.py:40]: `mypy` really expects here return type

Comment (stancld) [reply src/torchmetrics/functional/text/bert.py:40]: Maybe `contextlib.AbstractContextManager`

**Review by stancld — APPROVED** [review]

File: src/torchmetrics/functional/text/bert.py [hunk 5]
```diff
@@ -12,7 +12,9 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import csv
+import logging
 import urllib
+from contextlib import contextmanager
 from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union
```
Comment (Borda) [inline src/torchmetrics/functional/text/bert.py:18]:
```suggestion
from typing import Any, Callable, Dict, Iterator, List, Optional, Sequence, Tuple, Union
```

File: src/torchmetrics/functional/text/bert.py (hunk 4 above)
Comment (Borda) [reply src/torchmetrics/functional/text/bert.py:40]:
```suggestion
def _ignore_log_warning() -> Iterator[None]:
```
ref: https://stackoverflow.com/a/49736916/4521646

**Pull request merged**